### PR TITLE
Added support for switching the app theme without a restart

### PR DESCRIPTION
**Pull request by yaichenbaum**

### CODE REVIEW

File: Files/Helpers/ExternalResourcesHelper.cs [hunk 1]
```diff
@@ -117,12 +118,39 @@ public async Task<bool> TryLoadThemeAsync(AppTheme theme)
...
+            {
+                await TryLoadThemeAsync(NewTheme);
+            }
+        }
+
+        public bool RemoveTheme(AppTheme theme)
+        {
+            try
+            {
+                App.Current.Resources.MergedDictionaries.Remove(App.Current.Resources.MergedDictionaries.Where(x => x.TryGetValue("CustomThemeID", out var key) && (key as string) == theme.Key).FirstOrDefault());
```
Comment (duke7553) [inline Files/Helpers/ExternalResourcesHelper.cs:139]:
```suggestion
                App.Current.Resources.MergedDictionaries.Remove(App.Current.Resources.MergedDictionaries.FirstOrDefault(x => x.TryGetValue("CustomThemeID", out var key) && (key as string) == theme.Key));
```
Does something like this work in place of two LINQ calls?

File: Files/ViewModels/SettingsViewModels/AppearanceViewModel.cs [hunk 2]
```diff
@@ -107,16 +108,35 @@ public AppTheme SelectedTheme
...
-        public bool ShowRestartControl
+        /// <summary>
+        /// Forces the application to use the correct resource styles
+        /// </summary>
+        private async void UpdateTheme()
         {
-            get => showRestartControl;
-            set => SetProperty(ref showRestartControl, value);
+            // Allow time to remove the old theme
+            await Task.Delay(250);
```
Comment (duke7553) [inline Files/ViewModels/SettingsViewModels/AppearanceViewModel.cs:128]: Is there any way we can verify the old theme is removed without hardcoding in a delay?

File: Files/Helpers/ExternalResourcesHelper.cs (hunk 1 above)
Comment (yaichenbaum) [reply Files/Helpers/ExternalResourcesHelper.cs:139]: @winston-de

Comment (winston-de) [reply Files/Helpers/ExternalResourcesHelper.cs:139]: Yes it would, I didn't know that was possible thanks for pointing it out.

**Review by winston-de — APPROVED** [review]

**Review by duke7553 — APPROVED** [review]

**Pull request merged**